src/modules/phoneNumber/parsePhoneNumber.ts

```diff
@@ -0,0 +1,68 @@
+import { withErrorBoundary } from '../../common/errorBoundary';
+import { PHONE_FORMATTER_MAPPER } from './data/phoneFormatterMapper';
+import formatPhoneNumber from './formatPhoneNumber';
+import { detectCountryCodeFromDialCode, cleanPhoneNumber } from './utils';
+
+interface PhoneInfo {
+  countryCode: string;
+  dialCode: string;
+  formattedPhoneNumber: string;
+  formatTemplate: string;
+}
+
+// Parses a given phone number, identifies its country code (if not provided), and returns an object with details including the country code, formatted phone number, dial code, and format template.
+const parsePhoneNumber = (phoneNumber: string, country?: string): PhoneInfo => {
+  // Throw errors if phoneNumber is invalid
+  if (!phoneNumber) throw new Error('Parameter `phoneNumber` is invalid!');
+
+  // Clean the phoneNumber by removing non-numeric characters
+  phoneNumber = phoneNumber.toString();
+  phoneNumber = cleanPhoneNumber(phoneNumber);
+
+  // Detect or validate the country code
+  const countryCode =
```

Comment:
this countryCode is check is only present in parsePhoneNumber and absent in rest of the APIs.